server/store.go

```diff
@@ -547,8 +547,7 @@ func (sm *StoreMsg) clear() {
 	if sm == nil {
 		return
 	}
-	*sm = StoreMsg{_EMPTY_, nil, nil, sm.buf, 0, 0}
-	if len(sm.buf) > 0 {
-		sm.buf = sm.buf[:0]
-	}
+	sm.subj = _EMPTY_
```

Comment:
The advantage of the method above is that if StoreMsg{} has a new field, the code above would fail to compile until we update it to take care of the new field.
With your current approach, in that same situation, it requires someone to not forget to add that.

Reply:
This is true, although it will be more expensive to copy the full struct. I don’t know how much that would really matter in the real world.

Reply:
We could combine both approaches possibly.

```go
*sm = StoreMsg{_EMPTY_, sm.hdr[:0], ...}
```

But for this to be effective we would need to reuse any capacity there in the lower layers of filestore and memstore.

Also, I know `sm.hdr = sm.hdr[:0]` still works if original is nil but I did a double take at first.

Reply:
Agreed, I've made that change now.